catalog/app/utils/AWS/Bedrock/Bedrock.ts

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react'
+
+import BedrockRuntime from 'aws-sdk/clients/bedrockruntime'
+
+import cfg from 'constants/config'
+
+import * as Config from '../Config'
+import * as Credentials from '../Credentials'
+
+import { foldMessages, historyAppend, History } from './History'
+import { CONTENT_TYPE, bedrockBodyToMessage, createMessage } from './Message'
+
+const MODEL_ID = 'anthropic.claude-3-sonnet-20240229-v1:0'
+const ANTHROPIC_VERSION = 'bedrock-2023-05-31'
+
+// https://docs.aws.amazon.com/bedrock/latest/userguide/key-definitions.html
+// Token – A sequence of characters that a model can interpret or predict as a single unit of meaning.
+// Not a word, but eiher word, or phrase, or punctuaction mark, or word suffix or prefix.
+export const MAX_TOKENS = 100000
+
+// Bedrock calls are not free
+// https://aws.amazon.com/bedrock/pricing/
+//
+// You can use `MOCK_BEDROCK = true` to mock Bedrock API response
+const MOCK_BEDROCK = false
+
+export function useBedrock(
+  overrides?: Partial<BedrockRuntime.Types.ClientConfiguration>,
+) {
+  Credentials.use().suspend()
+
+  const awsConfig = Config.use()
+
+  const client: BedrockRuntime = React.useMemo(() => {
+    const opts = {
+      ...awsConfig,
+      region: cfg.region,
```

Comment:
afair it's not available in all regions, so it might make sense to use the default region here, just make sure it works ok when the stack is in a different region

Reply:
> *Claude 3 Opus is currently available in the US West (Oregon) Region

And other models available in many regions.

We enabled Bedrock only in us-east-1. That's why it doesn't work on default dev stack (which is in us-east-2).